File: slack-reader/src/main/java/dev/number6/slackreader/model/JoinChannelResponse.java
package dev.number6.slackreader.model;

public class JoinChannelResponse {

    private final boolean ok;

    public JoinChannelResponse(boolean ok) {
        this.ok = ok;
    }

    public static JoinChannelResponse failed() {
        return new JoinChannelResponse(false);
    }

    public static JoinChannelResponse ok() {
        return new JoinChannelResponse(true);
    }

    public boolean getOk() {
        return ok;
    }

    @Override
    public String toString() {
        return "JoinChannelResponse{" +
                "ok=" + ok +
                '}';
    }
}

File: slack-reader/src/main/java/dev/number6/slackreader/model/JoinChannelResponse.kt
package dev.number6.slackreader.model

class JoinChannelResponse(val ok: Boolean) {

    override fun toString(): String {
        return "JoinChannelResponse{" +
                "ok=" + ok +
                '}'
    }

    companion object {
        fun failed(): JoinChannelResponse {
            return JoinChannelResponse(false)
        }

        fun ok(): JoinChannelResponse {
            return JoinChannelResponse(true)
        }
    }
}